### PR TITLE
leave any server where the owner is blacklisted

### PR DESCRIPTION
**Pull request by Miky88**
Bot should instantly leave any server where the owner is blacklisted

### CODE REVIEW

**Comment by MoonlightCapital** [comment]
this won't work: you should leave every server inside a forEach. also remove the try/catch.

File: commands/blacklist.js [hunk 1]
```diff
@@ -27,6 +27,14 @@ exports.run = async (client, message, args) => {
     await client.db.updateUser(user.data)
 
     message.channel.send(`${greentick} ${client.utils.escapeMarkdown(user.tag)} (\`${user.id}\`) has been blacklisted`)
+    
+    //leave his server
+    try {
```
Comment (MoonlightCapital) [inline commands/blacklist.js:32]: Bet you know already it errored :kappalul:

File: commands/blacklist.js [hunk 2]
```diff
@@ -27,6 +27,14 @@ exports.run = async (client, message, args) => {
...
+      client.guilds.filter(c=>c.owner.id === user.id).leave()
```
Comment (MoonlightCapital) [inline commands/blacklist.js]: This will likely say `<Array>.leave() is not a function`. Put the leaves in a `forEach`.

**Review by MoonlightCapital — CHANGES_REQUESTED** [review]

File: commands/blacklist.js (hunk 2 above)
Comment (Miky88) [reply commands/blacklist.js]: ooo okkk

File: commands/blacklist.js (hunk 1 above)
Comment (Miky88) [reply commands/blacklist.js:32]: ew?

File: commands/blacklist.js [hunk 3]
```diff
@@ -30,7 +30,11 @@ exports.run = async (client, message, args) => {
     
     //leave his server
     try {
-      client.guilds.filter(c=>c.owner.id === user.id).leave()
+      if(client.guilds.filter(c=>c.owner.id === user.id) > 0) {
```
Comment (MoonlightCapital) [inline commands/blacklist.js]: This is pointless. If there are no servers it won't leave any.

Also, you're comparing an array with an integer.

File: commands/blacklist.js [hunk 4]
```diff
@@ -30,7 +30,11 @@ exports.run = async (client, message, args) => {
...
+        client.guilds.filter(c=>c.owner.id === user.id).forEach(g=>{
```
Comment (MoonlightCapital) [inline commands/blacklist.js]: Square brackets are not needed.

**Review by MoonlightCapital — CHANGES_REQUESTED** [review]

**Comment by Miky88** [comment]
i did it

File: commands/blacklist.js (hunk 3 above)
Comment (Miky88) [reply commands/blacklist.js]: i forgot .size... but ok removed

File: commands/blacklist.js (hunk 4 above)
Comment (NightScript370) [reply commands/blacklist.js]: What do you mean square brackets?

File: commands/blacklist.js [hunk 5]
```diff
@@ -27,6 +27,14 @@ exports.run = async (client, message, args) => {
     await client.db.updateUser(user.data)
 
     message.channel.send(`${greentick} ${client.utils.escapeMarkdown(user.tag)} (\`${user.id}\`) has been blacklisted`)
+    
+    //leave his server
+    try {
+        client.guilds.filter(c=>c.owner.id === user.id).forEach(g=>g.leave())
+    } catch (e){
+      console.log(e)
```
Comment (NightScript370) [inline commands/blacklist.js]: Why are you using console.log to log errors?
Either do it the correct way (console.error) or don't do it at all

Comment (Miky88) [reply commands/blacklist.js]: Better if I remove this part

Comment (NightScript370) [reply commands/blacklist.js]: I don't think it should be *removed*. It's always important to have callbacks
Why not just change it to the correct method?

Comment (Miky88) [reply commands/blacklist.js]: ok so...

**Comment by Miky88** [comment]
> also remove try/catch

hm why?

**Comment by NightScript370** [comment]
the heck?
I didn't tell you to remove try/catch.

Where are you pulling that quote from?

**Comment by Miky88** [comment]
moonlight said this

**Comment by Miky88** [comment]
first comment
>this won't work: you should leave every server inside a forEach. also remove the try/catc

**Comment by NightScript370** [comment]
Oh, I missed that. I was looking at the reviews, not the comments.
He doesn't like callbacks in code as it goes against his policy

**Comment by Miky88** [comment]
I leave it.. final decision goes to him

**Comment by MoonlightCapital** [comment]
So... what are we going to do with this?

**Comment by MoonlightCapital** [comment]
Closed due to inactivity.

**Pull request closed without merging**